bigquery/google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -0,0 +1,280 @@
+        'scale', 'null_ok',
+    ])
+
+
+class Cursor(object):
+    """DB-API Cursor to Google BigQuery.
+
+    :type: :class:`~google.cloud.bigquery.dbapi.Connection`
+    :param connection: A DB-API connection to Google BigQuery.
+    """
+    def __init__(self, connection):
+        self.connection = connection
+        self.description = None
+        # Per PEP 249: The attribute is -1 in case no .execute*() has been
+        # performed on the cursor or the rowcount of the last operation
+        # cannot be determined by the interface.
+        self.rowcount = -1
+        # Per PEP 249: The arraysize attribute defaults to 1, meaning to fetch
+        # a single row at a time.
+        self.arraysize = 1
+        self._query_data = None
+        self._page_token = None
+        self._has_fetched_all_rows = True
+
+    def close(self):
+        """No-op."""
+
+    def _set_description(self, schema):
+        """Set description from schema.
+
+        :type: Sequence[:class:`~google.cloud.bigquery.schema.SchemaField`]
+        :param schema: A description of fields in the schema.
+        """
+        if schema is None:
+            self.description = None
+            return
+
+        self.description = tuple([
+            Column(
+                name=field.name,
+                type_code=field.field_type,
+                display_size=None,
+                internal_size=None,
+                precision=None,
+                scale=None,
+                null_ok=field.mode == 'NULLABLE')
+            for field in schema])
+
+    def _set_rowcount(self, query_results):
+        """Set the rowcount from query results.
+
+        Normally, this sets rowcount to the number of rows returned by the
+        query, but if it was a DML statement, it sets rowcount to the number
+        of modified rows.
+
+        :type: :class:`~google.cloud.bigquery.query.QueryResults`
+        :param query_results: results of a query
+        """
+        total_rows = 0
+        num_dml_affected_rows = query_results.num_dml_affected_rows
+
+        if (query_results.total_rows is not None
+                and query_results.total_rows > 0):
+            total_rows = query_results.total_rows
+        if num_dml_affected_rows is not None and num_dml_affected_rows > 0:
+            total_rows = num_dml_affected_rows
+        self.rowcount = total_rows
+
+    def _format_operation_list(self, operation, parameters):
+        """Formats parameters in operation in the way BigQuery expects.
+
+        The input operation will be a query like ``SELECT %s`` and the output
+        will be a query like ``SELECT ?``.
+
+        :type: str
+        :param operation: A Google BigQuery query string.
+
+        :type: Sequence[Any]
+        :param parameters: Sequence of parameter values.
+        """
+        formatted_params = ['?' for _ in parameters]
+
+        try:
+            return operation % tuple(formatted_params)
+        except TypeError as ex:
```

Comment:
Pylint is currently turned off but it is coming back, and it will yell at you for a 2-char variable name. Might I suggest `exc`?